### DISCUSSION TITLE
http://lotabout.github.io/2016/write-a-C-interpreter-5/

### DISCUSSION THREAD

**Discussion by lotabout** [lotabout/lotabout.github.io#14] [Announcements]
# http://lotabout.github.io/2016/write-a-C-interpreter-5/
手把手教你构建 C 语言编译器（5）- 变量定义
http://lotabout.github.io/2016/write-a-C-interpreter-5/

**Comment by lotabout** [comment]
由 Bin Mu 发布于 2019-06-13T22:53:09Z：

终于把全部教程看完了，佩服不已。  
前面说"我们的语言不支持跨函数递归，而为了实现自举，实际上我们也不能使用递归"，这是为什么呢。我看见最后的例子就是斐波那契数列的递归程序啊

**Reply by lotabout** [reply]
由 lotabout 发布于 2019-06-15T03:19:45Z：

Bin Mu 语言不支持的是"跨函数"递归，即函数 a 调用函数 b，而函数 b 里又调用函数 a。所以为了自举，语法解析时也没有用到这种跨函数递归的形式。

**Reply by lotabout** [reply]
由 Bin Mu 发布于 2019-06-15T12:15:50Z：

好像明白了。如果有一个函数声明的机制，就没这个问题了吧。  
刚刚加了这样一个机制，虽然代码还不够健壮，但是现在可以交叉调用了。

**Comment by LeeeeeeM** [comment]
你写完自己能跑吗？
